app/(dashboard)/users/[id]/page.tsx

```diff
@@ -1,33 +1,46 @@
 import { User } from '@fiu-fit/common';
 import api from '@/api/serverSideAxiosConfig';
-import { UserCard } from '@/app/(dashboard)/users/components/UserCard';
-import UserDetailHeader from '@/app/(dashboard)/users/components/UserDetailHeader';
+import DetailCard from '@/components/DetailCard';
+import DetailHeader from '@/components/DetailHeader';
 
 async function getUser(id: number): Promise<User> {
   const { data: user } = await api.get<User>(`/users/${id}`);
 
   return user;
 }
 
-async function deleteUser(id: number): Promise<User> {
-  'use server';
-  const { data: user } = await api.delete<User>(`/users/${id}`);
-
-  return user;
-}
-
 export default async function UserDetail({
   params: { id },
 }: {
   params: { id: number };
 }) {
   const user = await getUser(id);
 
+  const deleteUser = async (): Promise<User> => {
+    'use server';
+    const { data: deletedUser } = await api.delete<User>(`/users/${id}`);
+
+    return deletedUser;
+  };
+
   return (
     <div className='w-full h-full'>
-      <UserDetailHeader user={user} deleteUser={deleteUser} />
-      <div className='mx-12 mt-12 w-1/2'>
-        <UserCard user={user} />
+      <div className='p-12 w-full'>
+        <DetailHeader
+          title={`${user.firstName} ${user.lastName}`}
+          onDelete={deleteUser}
+          afterDeleteRoute='/users'
+        />
+        <DetailCard
+          title='Detalle de usuario'
+          fields={{
+            ID:       user.id,
+            Email:    user.email,
+            Nombre:   user.firstName,
+            Apellido: user.lastName,
+            Rol:      user.role,
+          }}
```

Comment:
See other comment, extract this to a an object!

Reply:
Dale